indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -226,10 +227,11 @@ public Response taskPost(final Task task, @Context final HttpServletRequest req)
             taskQueue.add(task);
             return Response.ok(ImmutableMap.of("task", task.getId())).build();
           }
-          catch (DruidException e) {
-            return Response.status(e.getResponseCode())
-                           .entity(ImmutableMap.of("error", e.getMessage()))
-                           .build();
```

Comment:
You still need to retain this, unfortunately. The other DruidException could be caught here as well. You can remove the other deprecated DruidException in a follow-up clean up PR